.github/workflows/set-up-go/action.yaml

```diff
@@ -7,4 +7,4 @@ runs:
     - name: setup golang
       uses: actions/setup-go@v5
       with:
-        go-version: "1.21"
+        go-version: "1.22.0"
```

Comment:
@subhamkrai how about updating this to use go.mod so that we don't need to change in multiple places?